docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+3) If the module is found, it calls the `Module` object method `setGrade()` to update the grade field
+   and prints a success message by calling the `printUpdatedModuleGrade()` method of the `Print` class.
+4) Else, it prints an error message by calling the `printInvalidModule()` method of the `Print` class.
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![gradeCommand](diagrams/GradeCommand.png)
+
+<small><i>Figure ???</i></small>
+
+
+### 3.8. Calculate CAP
+
+`calculateCAP` command:
+
+The `calculateCAP` command is used by the user to calculate the Cumulative Average Point (CAP) of graded modules.
+
+The sequence by which the Parser class handles the `calculateCAP` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `calculateCAP()` method
+   in the `ModuleList` class.
+2) The method then loops through the moduleList array to find modules that have been graded.
+3) It checks if the module grade should be counted by self-invocation of the `shouldCountModule()` method
+4) If it is, proceeds to obtain the grade value by self-invocation of the `getGradeValue()` method.
+5) Calculates the required grade values, and finally calls the `printCalculatedCAP()` of the `Print` class to display
+   the result to the user
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![calculateCAP](diagrams/CalculateCAP.png)
+
+<small><i>Figure ???</i></small>
+
+## Appendix: Requirements
+
+###  Product Scope
+
+---
+
+### Target User Profile
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our target user profile is Information Security students in NUS.
 
+### User Stories
 
-## Product scope
-### Target user profile
+| Version | As a...      | I want to...                                    | So that I can...                                                               |
+|---------|--------------|-------------------------------------------------|--------------------------------------------------------------------------------|
+| v1.0    | New user     | See usage instructions                          | Refer to them when I forget the commands                                       |
+| v1.0    | New user     | See usage instructions based on feature         | Get more in depth knowledge on the feature that I require                      |
+| v1.0    | New user     | Find a module by its code                       | Locate the module code without having to go through the entire list of modules |
+| v2.0    | Current user | Find a module by its type                       | Locate all modules in a category without having to go through the entire list  |
+| v1.0    | New user     | Add modules                                     | To plan out my 4 years in uni                                                  |
```

Comment:
Perhaps can have a slightly more professional and better description for "So that I can"?